mets_mods2tei/api/mets.py

```diff
@@ -124,17 +126,74 @@ def __spur(self):
         """
 
         #
-        # main title and manuscript type
+        # get publication level
+        # get main and sub title from top-level logical div as a fallback
+        self.title = ""
+        self.biblevel = None
+        self.bibtype = None
         div = self.get_div_structure()
-        self.title = div.get_LABEL() if div else ""
-        self.type = div.get_TYPE() if div else ""
+        if div:
+            self.title = div.get_LABEL() # overridden by any titleInfo
+            div_type = div.get_TYPE()
+            # differentiate between analytic and closed, periodic and singular, dependent and indepenent types
+            # (for use in bibl/@type and biblFull//title/@level):
+            # FIXME: verify this ruleset is correct/standardized (but criteria do not look orthogonal, e.g. "issue" and "proceeding")
+            if div_type in ["bachelor_thesis", "diploma_thesis", "magister_thesis", "master_thesis", "doctoral_thesis", "habilitation_thesis", "file", "register", "research_paper", "report", "atlas", "album", "letter", "document", "leaflet", "manuscript", "poster", "plan", "study", "judgement", "preprint", "dossier", "paper"]:
+                self.biblevel = 'u' # unpublished
```

Comment:
Why `unpublished` for e.g. `doctoral_thesis`?

Reply:
Because among TEI's `a|m|j|s|u` it's the best fit for these.